ca_gazebo/src/model_pose_publisher_plugin.cpp

```diff
@@ -26,20 +33,24 @@ class ModelPosePublisherPlugin : public ModelPlugin
         return;
       }
       ROS_INFO("Model pose publisher started!");
-      ROS_INFO("model Name = %s", _parent->GetName().c_str());
 
       // Store the pointer to the model
-      this->model = _parent;
+      this->model_ = _parent;
+
+      ROS_INFO("model Name = %s", this->model_->GetName().c_str());
 
       // Listen to the update event. This event is broadcast every
       // simulation iteration.
       this->updateConnection = event::Events::ConnectWorldUpdateBegin(
           std::bind(&ModelPosePublisherPlugin::OnUpdate, this));
       this->prev_update_time_ = ros::Time::now();
 
-      this->rosnode_.reset(new ros::NodeHandle("ModelPose"));
-      this->pub_ = this->rosnode_->advertise<geometry_msgs::Pose>(("/ca_gazebo/" + _parent->GetName() + "/model_pose").c_str(), 100);
-      this->link = this->model->GetLink("link");
+      this->rosnode_.reset(new ros::NodeHandle());
+      this->pose_pub_ = this->rosnode_->advertise<geometry_msgs::Pose>((this->model_->GetName() + "/pose").c_str(), 1);
+      this->odom_pub_ = this->rosnode_->advertise<nav_msgs::Odometry>((this->model_->GetName() + "/odom").c_str(), 1);
+      this->link_ = this->model_->GetLink("link");
+
+      this->odom_msg_.header.frame_id = "/map";
```

Comment:
@eborghi10 this may work because of backwards compatibility support (see [here](http://wiki.ros.org/tf2/Migration), section 1.7), but leading slashes are banned from frame IDs in `tf2`(see [here](http://wiki.ros.org/tf2/Design#Multi-Robot_Support), section 3.1). Consider just using `"map"`.

Reply:
You're right. `map` should work. I checked the name so I'll work on it.